### PR TITLE
empty!() for CircularDeque, add it to docs, add some other missing things to docs

### PR DESCRIPTION
**Pull request by KadeG**

### CODE REVIEW

File: src/circular_buffer.jl [hunk 1]
```diff
@@ -9,6 +9,11 @@ type CircularBuffer{T} <: AbstractVector{T}
     (::Type{CircularBuffer{T}}){T}(capacity::Int) = new{T}(capacity, 1, T[])
 end
 
+function Base.empty!(cb::CircularBuffer)
+    cb.buffer = typeof(cb.buffer)[]
+    cb
+end
```
Comment (kmsquire) [inline src/circular_buffer.jl]: I think you also need to reset `cb.first = 1`.  Is there a reason you didn't just call `empty!(cb.buffer)`?

Comment (kmsquire) [reply src/circular_buffer.jl]: Also, could you add a quick test for this?

Comment (KadeG) [reply src/circular_buffer.jl]: Changes make. Test added. You're suggestion is much faster. Thanks for the help!

**Comment by kmsquire** [comment]
Coverage reduction is a problem in Coverage.jl: https://github.com/JuliaCI/Coverage.jl/issues/132

LGTM!  Thanks for the contribution.

**Pull request merged**